lib/stateoscope/visualizer.rb

```diff
@@ -10,8 +10,11 @@ def parse_graph
       add_state_transitions
     end
 
-    def output(filename)
-      @viz.output(pdf: "#{filename}.pdf")
+    def output(filename, format)
+      options = {}
+      options[format] = "#{filename}.#{format}"
+      @viz.output(options)
+      "#{filename}.#{format}"
```

Comment:
Why is this line necessary?

Reply:
I'm using Stateoscope not through rake but dynamically in a Rails project. In one of my controllers I do

```
filepath = Stateoscope.visualize(
        BaustarterState,
        current_state: @user.baustarter_state.state,
        dir: Rails.root.join("public"),
        format: "png"
      )

# return filepath to client
```

Anyway since the filename is created from a timestamp in `Stateoscope.filename_for` it would be nice to return the final filename to the caller, otherwise they have no idea where the just created file is located.

Reply:
ok yea I removed it from the visualizer and just return the filename from `visualize` method. Better.